static/js/pages/applications/ApplicationDashboardPage.js

```diff
@@ -445,9 +455,15 @@ export class ApplicationDashboardPage extends React.Component<Props, State> {
 
     return (
       <div className="p-3 mt-3 application-detail">
-        {profileRow}
-        {resumeRow}
-        {submissionStepRows}
+        {currentUser.profile &&
+        currentUser.profile.can_skip_application_steps === true &&
+        application.bootcamp_run.allows_skipped_steps === true ? null : (
+            <Fragment>
+              {profileRow}
+              {resumeRow}
+              {submissionStepRows}
+            </Fragment>
+          )}
```

Comment:
If you implement the helper function I recommended above, this can be refactored... 

```suggestion
        {!isEligibleToSkipSteps(currentUser, application) && (
            <Fragment>
              {profileRow}
              {resumeRow}
              {submissionStepRows}
            </Fragment>
          )}
```